### PR TITLE
Cbbf 93 scalable ec2 instances

### PR DESCRIPTION
**Pull request by glovercloud**
Tested within bb-fhir AWS environment and working against the devcloud sandbox.  And yes that is with a proxy specified.  That is the only way the instances can reach the world at the moment.  Easy to change once the test sandbox is operational.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,23 +5,84 @@ set of load and performance tests for the FHIR backend test server.
...
 ## Requirements
 
-This test suite has dependencies on the bluebutton-server application.  Clone
+### Bluebutton-server Application
+
+This test suite has dependencies on the bluebutton-data-server application.  Clone
 that repo from: 
 
-https://github.com/HHSIDEAlab/bluebutton-server
+https://github.com/HHSIDEAlab/bluebutton-data-server
```
Comment (karlmdavis) [inline README.md]: Probably want to update the GH org to "`CMSgov`". No big deal, though.

File: ansible/ansible.cfg [hunk 2]
```diff
@@ -4,5 +4,9 @@ vault_password_file = vault.password
 # Automatically accept SSH thumbprints when machines are provisioned.
 # ATTENTION: DO NOT USE in production environments as it allows for man-in-the-middle attacks
 host_key_checking = False
+remote_user = ec2-user
 
-inventory = ./hosts.yml
+# Tweak some timeout settings to try and prevent ssh failures
```
Comment (karlmdavis) [inline ansible/ansible.cfg:9]: Sorry I haven't made my way in my inbox to your "SSH timeouts" question. I've run into that bunches, too. Things that help:

1. Add an artificial pause after EC2 instances are stood up in your "wait for SSH" task, e.g. the `delay` here: https://github.com/CMSgov/bluebutton-ansible-playbooks-data-sandbox/blob/master/etl_server_redeploy.yml#L98
2. Always, always, always use ansible pipelining:
    * https://github.com/CMSgov/bluebutton-ansible-playbooks-data-sandbox/blob/master/etl_server_redeploy.yml#L112
    * https://github.com/CMSgov/bluebutton-ansible-playbooks-data-sandbox/tree/master/roles/rhel_ansible_pipelining
    * https://github.com/CMSgov/bluebutton-ansible-playbooks-data-sandbox/blob/master/ansible.cfg#L23
3. Bump Ansible's `timeout`: https://github.com/CMSgov/bluebutton-ansible-playbooks-data-sandbox/blob/master/ansible.cfg#L20

File: ansible/fhir-stress-test.yml [hunk 3]
```diff
@@ -19,87 +24,112 @@
...
+        url: "http://archive.apache.org/dist/jmeter/binaries/apache-jmeter-{{ jmeter_version }}.tgz"
+        dest: "support/apache-jmeter-{{ jmeter_version }}.tgz"
+
+    - name: Acquire a session token
+      sts_session_token:
+        profile: "{{ profile }}"
+        region: "{{ region }}"
+        mfa_serial_number: "{{ mfa_sn }}"
+        mfa_token: "{{ mfa }}"
+      register: session_credentials
```
Comment (karlmdavis) [inline ansible/fhir-stress-test.yml:48]: This totally works for now, but keep in mind that if/when we run all this from Jenkins, it won't be needed from there. Jenkins will instead leverage the permissions assigned to its EC2 instance profile.

File: ansible/fhir-stress-test.yml [hunk 4]
```diff
@@ -19,87 +24,112 @@
...
+
+    - name: EC2 - Provision 'bluebutton-stress' client
+      ec2:
+        aws_access_key: "{{ session_credentials.sts_creds.access_key }}"
+        aws_secret_key: "{{ session_credentials.sts_creds.secret_key }}"
+        security_token: "{{ session_credentials.sts_creds.session_token }}"
```
Comment (karlmdavis) [inline ansible/fhir-stress-test.yml:54]: These shouldn't be needed on any tasks, if your `~/.aws/credentials` is configured correctly.

**Review by karlmdavis — APPROVED** [review]

File: ansible/ansible.cfg (hunk 2 above)
Comment (karlmdavis) [reply ansible/ansible.cfg:9]: (I probably made it sound like those aren't complete solutions. They actually are, as far as I can tell: the `-sandbox` plays never seem to run into SSH timeouts anymore for me, with those all in place. I d still have a n intermittent problem with YUM being locked/unavailable after instances start, but it's rare enough that I haven't gone and tracked it down yet.)

File: ansible/fhir-stress-test.yml (hunk 4 above)
Comment (glovercloud) [reply ansible/fhir-stress-test.yml:54]: These are temporary credentials that are created within the script.  How do I configure aws/credentials to use something created within ansible?

File: ansible/fhir-stress-test.yml (hunk 3 above)
Comment (glovercloud) [reply ansible/fhir-stress-test.yml:48]: Understood, but I needed to test it! 👍

**Review by jcgeer — APPROVED** [review]

**Review by dho937 — APPROVED** [review]

**Pull request merged**